### PR TITLE
[fix][OSS] enabling disabled tests for 1.8

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fixes part of #516. The tests were wrong in that the cuda device was not specified, that's why it hanged

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/optim/test_oss.py [hunk 1]
```diff
@@ -472,24 +473,21 @@ def closure():
     dist.destroy_process_group()
 
 
-# TODO(blefaudeux) Fix for torch v1.8.0
-@pytest.mark.skipif(torch.__version__.split("+")[0].split(".") == ["1", "8", "0"], reason="disabled for torch 1.8.0")
+@skip_if_single_gpu
 def test_collect_shards():
-    world_size = 3
```
Comment (blefaudeux) [inline tests/optim/test_oss.py:478]: 3 was arbitrary, I think it was a copy paste from another test (multi groups) which required that

File: tests/optim/test_oss.py [hunk 2]
```diff
@@ -472,24 +473,21 @@ def closure():
...
+    world_size = 2
     temp_file_name = tempfile.mkstemp()[1]
-
-    if torch.cuda.is_available():
```
Comment (blefaudeux) [inline tests/optim/test_oss.py:481]: this is already guarded by the decorator

File: tests/optim/test_oss.py [hunk 3]
```diff
@@ -472,24 +473,21 @@ def closure():
...
     mp.spawn(
         run_test_collect_shards, args=(world_size, reference_rank, temp_file_name), nprocs=world_size, join=True,
     )
 
 
-def run_test_reproducibility(rank, world_size, reference_rank, tempfile_name):
+def run_test_reproducibility(rank, world_size, tempfile_name):
     dist_init(rank, world_size, tempfile_name)
     device = torch.device(rank) if torch.cuda.device_count() > 1 else DEVICE
+    torch.cuda.set_device(rank)
```
Comment (blefaudeux) [inline tests/optim/test_oss.py:490]: this was the reason why it would hang when communicating

File: tests/optim/test_oss.py [hunk 4]
```diff
@@ -415,6 +415,7 @@ def test_sharding():
 def run_test_collect_shards(rank, world_size, reference_rank, tempfile_name):
     dist_init(rank, world_size, tempfile_name)
     device = torch.device(rank) if torch.cuda.device_count() > 1 else DEVICE
+    torch.cuda.set_device(rank)
```
Comment (blefaudeux) [inline tests/optim/test_oss.py:418]: same

**Review by anj-s — APPROVED** [review]

**Review by min-xu-ai — COMMENTED** [review]
nice

**Pull request merged**